### ISSUE TITLE
Iterating over semi local elements fails on refined meshes

### ISSUE THREAD

**Issue by dschwen** [libMesh/libmesh#974]
Incrementing the iterator obtained via `.semilocal_elements_begin()` can trigger an `Assertion \`start_elem->active()' failed. `

**Comment by dschwen** [comment]
@jwpeterson has proposed a fix for this: adding

```
this->_predicates.push_back(new active<T>)
```

to the `SemiLocal` multi-predicate. This seems to fix teh issue I observe.

**Comment by jwpeterson** [comment]
Will make PR momentarily...

**Comment by roystgnr** [comment]
Semilocal elements are not necessary active elements, and we may already have other code in which that assumption is important.  The fix here should probably be two-fold: create a semilocal_active_elements_foo() pair of iterators if that's what you need, and fix the iteration in the semilocal_elements_foo() iterators too.

**Comment by roystgnr** [comment]
Huh.  We do rely on find_point_neighbors(), which does assert active().  This seems like the underlying bug...

**Comment by jwpeterson** [comment]
OK, so you think that `find_point_neighbors()` shouldn't assert this?

**Comment by roystgnr** [comment]
No, find_point_neighbors() was definitely written to work for active elements only.  Perhaps what we need to be doing is, for inactive elements, looping over the children and testing is_semilocal() on each of them?

**Comment by roystgnr** [comment]
Yeah, that sounds like the right fix - iff any descendant of mine is semilocal, then so am I.  Could be slow as hell to iterate through the low level ancestors... but I suppose the number of remote ancestors decreases exponentially with level.  And I don't think there's any other safe way to handle it.

**Comment by jwpeterson** [comment]
OK, is there any reason to have #974 as a temporary quick fix until someone implements this loop over children checking for semi-locality?  Otherwise I will just close that...
